examples/providers/AllClustersExampleDeviceInfoProviderImpl.cpp

```diff
@@ -0,0 +1,254 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <AllClustersExampleDeviceInfoProviderImpl.h>
+
+#include <lib/core/TLV.h>
+#include <lib/support/CHIPMemString.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/DefaultStorageKeyAllocator.h>
+#include <lib/support/SafeInt.h>
+#include <lib/support/Span.h>
+#include <platform/internal/CHIPDeviceLayerInternal.h>
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <cstring>
+
+namespace chip {
+namespace DeviceLayer {
+
+namespace {
+constexpr TLV::Tag kLabelNameTag  = TLV::ContextTag(0);
+constexpr TLV::Tag kLabelValueTag = TLV::ContextTag(1);
+} // anonymous namespace
+
+AllClustersExampleDeviceInfoProviderImpl & AllClustersExampleDeviceInfoProviderImpl::GetDefaultInstance()
+{
+    static AllClustersExampleDeviceInfoProviderImpl sInstance;
+    return sInstance;
+}
+
+// See below for an example of FixedLabel and UserLabel setter/iterator.
+// WARNING: DO NOT USE THESE DEFAULT IMPLEMENTATIONS WITH DEFAULT VALUES
+// IN PRODUCTION PRODUCTS WITHOUT AUDITING THEM!
+//
+// The FixedLabel cluster, if used, should have values that have been vetted
+// for correctness in the product.
+DeviceInfoProvider::FixedLabelIterator * AllClustersExampleDeviceInfoProviderImpl::IterateFixedLabel(EndpointId endpoint)
+{
+    return chip::Platform::New<FixedLabelIteratorImpl>(endpoint);
+}
+
+AllClustersExampleDeviceInfoProviderImpl::FixedLabelIteratorImpl::FixedLabelIteratorImpl(EndpointId endpoint) : mEndpoint(endpoint)
```

Comment:
Is mEndpoint used anywhere?  I know this just got copied....

Reply:
Yes it is in the iterator for AllClustersExampleXxxxx

Reply:
It's used for UserLabel, but not FixedLabel.  For FixedLabel it's just stored in a member, then ignored, no?

Reply:
It's "used" through the iterations since the iterator keeps state of its position between calls and has to remember which endpoint it's for. It's just that the sample did not fully take care of exemplifying how to provide different data for different endpoints :( So it's not used, but should be used, and I don't want to take it out in case someone cuts and pastes it some more.